### ISSUE TITLE
Registry always uses the master branch Dockerfile

### ISSUE THREAD

**Issue by Gurpartap** [docker/docker-registry#550]
tldr; registry always uses Dockerfile from master branch even when tag is specified.

I created a [new Github repo](https://github.com/Gurpartap/docker-ruby-build) for ruby-build Dockerfile with a master branch and several tags.

And then I created [an auto build repo](https://registry.hub.docker.com/u/gurpartap/ruby-build) on Docker Registry, attached with the above Github repo. Mapped a couple of tags as shown in the screenshot:

![](https://www.dropbox.com/s/0gzyy94bo772d72/Screen%20Shot%202014-08-31%20at%205.44.21%20am.png?dl=1)

However the builds fail. Build details reveal that they are still using the Dockerfile from the master branch, and not the tag.

This is a service breaking bug. And I couldn't find any other place to report this.
